### DISCUSSION TITLE
whereHas not loading in final query

### DISCUSSION THREAD

**Discussion by kvnsya** [rappasoft/laravel-livewire-tables#1725] [Q&A]
Hi guys,

I have a hasOne relation with a whereHas in. 
But laravel-livewire-tables skips the whereHas if i check the query.

Is there a way to make this work? 

```
public function builder(): Builder
{
 return Article::query();
}

public function columns(): array
{
  return [
        Column::make('quantity', 'cartLine.quantity')


class Article extends Model
{
    public function cartLine(): HasOne
    {
        return $this->hasOne(CartLine::class, 'article_id')

                 ////////////// this part is not in the final query /////////////////
            ->whereHas('cart', function ($query) {
                $query->where('user_id', auth()->id());
            });
    }


//////////////// This is the query but no whereHas /////////////

 select `articles`.`id` as `id`, `cartLine`.`quantity` as `cartLine.quantity`, `articles`.`price` as `price` from `articles` left join `cart_lines` as `cartLine` on `cartLine`.`article_id` = `articles`.`id` limit 10 offset 0
```

Thanks a lot for you time!

**Comment by kvnsya** [comment]
Update: 

I now have this query and if i dump the data the quantity is in the Article data.

```
        $query = Article::with(['cartLine' => function ($query) {
            $query->whereHas('cart', function ($query) {
                            $query->where('user_id', auth()->id());
            });
        }])
            ->where('category', $this->category)
            ->select('articles.*')
            ->addSelect(['quantity' => CartLine::select('quantity')
                ->whereColumn('articles.id', 'cart_lines.article_id')
                ->join('carts', 'cart_lines.cart_id', '=', 'carts.id')
                ->where('carts.user_id', auth()->id())
                ->limit(1)
            ])
            ->orderByDesc('quantity');
```

But if i try to load it here i get Unknown column 'articles.quantity' in 'field list'

         Column::make('quantity', 'quantity')
                ->sortable()
                ->collapseOnMobile(),

Is there any way to fix this or other way to do this?

**Comment by lrljoe** [comment]
> Update: 
> 
> I now have this query and if i dump the data the quantity is in the Article data.
> 
> ```
>         $query = Article::with(['cartLine' => function ($query) {
>             $query->whereHas('cart', function ($query) {
>                             $query->where('user_id', auth()->id());
>             });
>         }])
>             ->where('category', $this->category)
>             ->select('articles.*')
>             ->addSelect(['quantity' => CartLine::select('quantity')
>                 ->whereColumn('articles.id', 'cart_lines.article_id')
>                 ->join('carts', 'cart_lines.cart_id', '=', 'carts.id')
>                 ->where('carts.user_id', auth()->id())
>                 ->limit(1)
>             ])
>             ->orderByDesc('quantity');
> ```
> 
> But if i try to load it here i get Unknown column 'articles.quantity' in 'field list'
> 
>          Column::make('quantity', 'quantity')
>                 ->sortable()
>                 ->collapseOnMobile(),
> 
> Is there any way to fix this or other way to do this? 
> 

Are you still having an issue?
